### PR TITLE
Support built in method calls for known Objects in cnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/cnex/global.c [hunk 1]
```diff
@@ -1218,6 +1218,46 @@ void object__invokeMethod(TExecutor *exec)
...
 
+    if (obj->object->type == oArray) {
+        if (strcmp(name, "size") == 0) {
+            if (args->array != NULL && args->array->size != 0) {
+                exec->rtl_raise(exec, "DynamicConversionException", "invalid number of arguments to size() (expected 0)");
+                goto cleanup;
+            }
+            push(exec->stack, cell_fromObject(object_createNumberObject(number_from_uint64(((Cell*)obj->object->ptr)->array->size))));
+            goto cleanup;
+        }
```
Comment (ghewgill) [inline exec/cnex/global.c:1229]: Need an `else` clause here to catch unknown method names and raise an exception.

File: exec/cnex/global.c [hunk 2]
```diff
@@ -1218,6 +1218,46 @@ void object__invokeMethod(TExecutor *exec)
...
+                Cell *key = cell_fromObject(object_createStringObject(keys->array->data[i].string));
+                array_appendElement(okeys, key);
+                cell_freeCell(key);
+            }
+            Cell *oArray = cell_fromObject(object_createArrayObject(okeys));
+            push(exec->stack, oArray);
+            cell_freeCell(keys);
+            array_freeArray(okeys);
+            goto cleanup;
+        }
```
Comment (ghewgill) [inline exec/cnex/global.c:1258]: Also handle unknown method names here.

File: exec/cnex/global.c [hunk 3]
```diff
@@ -1218,6 +1218,46 @@ void object__invokeMethod(TExecutor *exec)
...
+                goto cleanup;
+            }
+            push(exec->stack, cell_fromObject(object_createNumberObject(number_from_sint64(((Cell*)obj->object->ptr)->dictionary->len))));
+            goto cleanup;
+        } else if (strcmp(name, "keys") == 0) {
+            if (args->array != NULL && args->array->size != 0) {
+                exec->rtl_raise(exec, "DynamicConversionException", "invalid number of arguments to keys() (expected 0)");
+                goto cleanup;
+            }
+            // ToDo: Consider doing an "inplace" conversion, where the "keys" Cell is manually converted into an Object Array.
```
Comment (ghewgill) [inline exec/cnex/global.c:1245]: What does this comment mean? What would be the benefit of changing the way this works now?

**Review by ghewgill — CHANGES_REQUESTED** [review]

File: exec/cnex/global.c (hunk 3 above)
Comment (gitlarryf) [reply exec/cnex/global.c:1245]: That was a mental note to myself, to see if I could maybe keep the same Cell Array, but convert all the members to Object Strings, from TStrings, without having to build a whole new array.  I wanted to write the code to try it out, but then realized I was optimizing a problem I didn't even know I had; so, I left it as a comment, and walked away.

File: exec/cnex/global.c (hunk 2 above)
Comment (gitlarryf) [reply exec/cnex/global.c:1258]: Fixed.

File: exec/cnex/global.c (hunk 1 above)
Comment (gitlarryf) [reply exec/cnex/global.c:1229]: Fixed.

**Pull request merged**